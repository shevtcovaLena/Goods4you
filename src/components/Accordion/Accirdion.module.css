@import url("../../styles/variables.css");

.accordion {
  color: var(--text-secondary);
}

.title {
  font-size: 20px;
  font-weight: normal;
  line-height: 30px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 20px 0;
  cursor: pointer;
  transition: background-color 0.5s;
}

.content {
  font-size: 16px;
  font-weight: 600;
  line-height: 26px;
  max-height: 0;
  overflow: hidden;
  transition: max-height 0.5s ease-out;
}

.content p {
    margin-bottom: 20px;
}

.content.open {
  max-height: 300px;
  transition: max-height 0.5s ease-in;
}

.icon {
  width: 20px;
  height: 20px;
  background-image: url("/src/assets/+.svg");
  background-size: contain;
  background-repeat: no-repeat;
  transition: transform 0.5s;
}

.open.icon {
  transform: rotate(45deg);
}
