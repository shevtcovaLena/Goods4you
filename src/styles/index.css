@import url("./variables.css");

@font-face {
  font-family: "Jost";
  src: url("/src/assets/fonts/Jost-ExtraBold.woff2") format("woff2"),
    url("/src/assets/fonts/Jost-ExtraBold.woff") format("woff");
  font-weight: 800;
  font-style: normal;
  unicode-range: U+0020-007F, U+20A0-20CF;
  font-display: swap;
}

@font-face {
  font-family: "Jost";
  src: url("/src/assets/fonts/Jost-Bold.woff2") format("woff2"),
    url("/src/assets/fonts/Jost-Bold.woff") format("woff");
  font-weight: bold;
  font-style: normal;
  unicode-range: U+0020-007F, U+20A0-20CF;
  font-display: swap;
}

@font-face {
  font-family: "Jost";
  src: url("/src/assets/fonts/Jost-SemiBold.woff2") format("woff2"),
    url("/src/assets/fonts/Jost-SemiBold.woff") format("woff");
  font-weight: 600;
  font-style: normal;
  unicode-range: U+0020-007F, U+20A0-20CF;
  font-display: swap;
}

@font-face {
  font-family: "Jost";
  src: url("/src/assets/fonts/Jost-Medium.woff2") format("woff2"),
    url("/src/assets/fonts/Jost-Medium.woff") format("woff");
  font-weight: 500;
  font-style: normal;
  unicode-range: U+0020-007F, U+20A0-20CF;
  font-display: swap;
}

@font-face {
  font-family: "Jost";
  src: url("/src/assets/fonts/Jost-Regular.woff2") format("woff2"),
    url("/src/assets/fonts/Jost-Regular.woff") format("woff");
  font-weight: normal;
  font-style: normal;
  unicode-range: U+0020-007F, U+20A0-20CF;
  font-display: swap;
}

* {
  padding: 0px;
  margin: 0px;
  border: none;
  box-sizing: border-box;
  font-family: "Jost", sans-serif;
}

ul,
ol {
  margin: 0;
  padding: 0;
  list-style: none;
}

a {
  text-decoration: none;
  color: inherit;
}

a:hover {
  text-decoration: underline;
}

li {
  margin: 0;
  padding: 0;
}

:root {
  font-weight: 400;
  color: var(--text-primary);
  background-color: var(--bg-primary);

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#root {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}

footer,
header {
  width: 100%;
  height: 90px;
  color: var(--text-secondary);
  display: flex;
  justify-content: center;
  align-items: center;
}

header {
  background-color: var(--bg-header);
}

footer {
  background-color: var(--bg-footer);
}

main {
  width: 100%;
  flex-grow: 1;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.container {
  max-width: 1180px;
  width: 100%;
  margin: 0 10px;
}

input {
  width: 100%;
  background-color: #F6F6F6;
  height: 62px;
  font-size: 16px;
  line-height: 22px;
  border-radius: 4px;
  padding: 20px;
}

input::placeholder {
  color: #B2B5BB;
}

input:focus {
  outline: none;
}

header .container,
footer .container {
  display: flex;
  justify-content: space-between;
}

#logo {
  font-size: 30px;
  font-weight: 700;
  line-height: 44px;
}

#logo:hover {
  text-decoration: none;
}

nav,
ul {
  display: flex;
  align-items: center;
  gap: 40px;
}

li {
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
}

.link-with-icon {
  display: flex;
  align-items: center;
  gap: 6px;
}

.icon {
  position: relative;
  overflow: visible;
  display: inline-block;
  height: 20px;
}

.circle-badge {
  position: absolute;
  height: 12px;
  width: 12px;
  right: -4px;
  bottom: -4px;
  background-color: var(--accent);
  font-size: 8px;
  font-weight: 400;
  line-height: 8px;
  color: var(--text-secondary);
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  text-align: center;
}

.loader{
  border-radius: 50%;
  display: inline-block;
  border-top: 3px solid rgb(173, 173, 173);
  border-right: 3px solid transparent;
  animation: rotation 2s linear infinite;
}

.loader-box{
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

@keyframes rotation {
  0% { transform: rotate(0deg) }
  100% { transform: rotate(360deg) }
}

h1 {
  font-size: 36px;
  line-height: 56px;
}

nav li:hover a,
footer li:hover a {
  color: #c4c4c4;
}

@media (max-width: 540px) {
  header .container {
    justify-content: center;
    gap: 10px;
    flex-wrap: wrap;
  }

  nav,
  ul {
    gap: 30px;
  }
}
