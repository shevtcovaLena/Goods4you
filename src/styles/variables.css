:root {
  --bg-primary: #ffffff;
  --bg-header: #484283;
  --bg-footer: #444b58;
  --text-primary: #444b58;
  --text-secondary: #ffffff;
  --accent: #F14F4F;
}
