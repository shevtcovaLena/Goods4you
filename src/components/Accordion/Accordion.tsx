import React, { useState } from "react";
import styles from "./Accirdion.module.css";

interface AccordionItemProps {
  title: string;
  children: React.ReactNode;
}

export function AccordionItem({ title, children }: AccordionItemProps) {
  const [isOpen, setIsOpen] = useState(false);

  const toggleAccordion = () => {
    setIsOpen(!isOpen);
  };

  return (
    <div className={styles.accordion}>
      <div className={styles.title} onClick={toggleAccordion}>
        {title}
        <span className={`${styles.icon} ${isOpen ? styles.open : ""}`}></span>
      </div>
      <div className={`${styles.content} ${isOpen ? styles.open : ""}`}>
        <p>{children}</p>
      </div>
    </div>
  );
}
