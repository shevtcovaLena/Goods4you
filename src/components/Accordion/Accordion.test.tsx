import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { AccordionItem } from './Accordion';
import styles from './Accirdion.module.css'; // Импортируем стили для проверки

// Мок данных
const title = 'Accordion Title';
const content = 'Accordion Content';

xdescribe('AccordionItem component', () => {
  test('renders with the correct title and content', () => {
    render(<AccordionItem title={title}>{content}</AccordionItem>);

    // Проверяем, что заголовок отображается
    const titleElement = screen.getByText(title);
    expect(titleElement).toBeInTheDocument();

    // Проверяем, что содержимое по умолчанию скрыто
    const contentElement = screen.getByText(content);
    expect(contentElement).toBeInTheDocument(); // Содержимое должно быть в DOM
    expect(contentElement).not.toBeVisible(); // Проверяем, что содержимое скрыто
  });

  test('displays content when clicked', () => {
    render(<AccordionItem title={title}>{content}</AccordionItem>);

    // Находим элемент заголовка и кликаем по нему
    const titleElement = screen.getByText(title);
    fireEvent.click(titleElement);

    // Проверяем, что содержимое стало видимым
    const contentElement = screen.getByText(content);
    expect(contentElement).toBeVisible(); // Проверяем, что содержимое видно
  });

  test('hides content when clicked again', () => {
    render(<AccordionItem title={title}>{content}</AccordionItem>);

    // Находим элемент заголовка и кликаем по нему дважды
    const titleElement = screen.getByText(title);
    fireEvent.click(titleElement); // Открыть
    fireEvent.click(titleElement); // Закрыть

    // Проверяем, что содержимое снова скрыто
    const contentElement = screen.getByText(content);
    expect(contentElement).not.toBeVisible(); // Проверяем, что содержимое снова скрыто
  });

  test('applies open styles when open', () => {
    const { container } = render(<AccordionItem title={title}>{content}</AccordionItem>);

    // Находим элемент заголовка и кликаем по нему
    const titleElement = screen.getByText(title);
    fireEvent.click(titleElement);

    // Проверяем, что элемент с контентом имеет открытые стили
    const contentElement = container.querySelector(`.${styles.content}`);
    expect(contentElement).toHaveClass(styles.open); // Проверяем, что содержимое имеет класс open
  });

  test('applies icon rotation when open', () => {
    const { container } = render(<AccordionItem title={title}>{content}</AccordionItem>);

    // Находим элемент заголовка и кликаем по нему
    const titleElement = screen.getByText(title);
    fireEvent.click(titleElement);

    // Проверяем, что иконка повернута
    const iconElement = container.querySelector(`.${styles.icon}`);
    expect(iconElement).toHaveClass(styles.open); // Проверяем, что иконка имеет класс open
  });
});
