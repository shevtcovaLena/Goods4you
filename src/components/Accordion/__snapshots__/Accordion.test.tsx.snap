// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AccordionItem and matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Accordion Title
      <span
        class="undefined "
      />
    </div>
    <div
      class="undefined "
    >
      <p>
        Accordion Content
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders AccordionItem and matches snapshot 2`] = `
<DocumentFragment>
  <div>
    <div>
      Accordion Title
      <span
        class="undefined undefined"
      />
    </div>
    <div
      class="undefined undefined"
    >
      <p>
        Accordion Content
      </p>
    </div>
  </div>
</DocumentFragment>
`;
